packages/vite-plugin/src/main.ts

```diff
@@ -315,7 +315,10 @@ export default ({
       }
 
       if (ext === '.meta') {
-        const sourceFile = viteConfig.command === 'build' ? false : `'${id}'`;
+        const sourceFile =
+          viteConfig.command === 'build'
+            ? false
+            : `'${id.replace(/'/g, "\\'")}'`;
```

Comment:
I think doing this would be a bit more robust:
```suggestion
            : JSON.stringify(id);
```
It should automatically handle things like new lines and other strange character people may have in their paths.